### PR TITLE
Removing CMake warning for zero_copy_data_transfer

### PR DESCRIPTION
**Pull request by tyoungsc**
# Existing Sample Changes
## Description
Removing 'Manually-specified variables were not used by the project' warning from CMake

## Type of change
- [X] removing warnings

## How Has This Been Tested?
- [X] Command Line

### CODE REVIEW

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +21,15 @@ endif()
 # to ensure the BSP has the required support. Allow the user to define USM_HOST_ALLOCATIONS_ENABLED
 # to override this check (e.g., cmake .. -DUSM_HOST_ALLOCATIONS_ENABLED=1)
 if(NOT FPGA_BOARD MATCHES ".usm.*" AND NOT DEFINED USM_HOST_ALLOCATIONS_ENABLED)
-    message(FATAL_ERROR "This tutorial requires a BSP that has USM host allocations enabled")
+    message(FATAL_ERROR "This tutorial requires a BSP that has USM host allocations enabled.")
+endif()
+
+if (DEFINED USM_HOST_ALLOCATIONS_ENABLED)
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt:27]: These are meant to be bools, right?  Why not just if (USM_HOST_ALLOCATIONS_ENABLED) ?  That way you don't have to print the value in order to 'use' this variable and avoid the warning.  I think we do this in other samples.

Comment (tyoungsc) [reply DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt:27]: Done

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt [hunk 2]
```diff
@@ -20,16 +20,16 @@ endif()
 # this tutorial requires USM host allocations. Check the BSP name (which should contain the text 'usm')
 # to ensure the BSP has the required support. Allow the user to define USM_HOST_ALLOCATIONS_ENABLED
 # to override this check (e.g., cmake .. -DUSM_HOST_ALLOCATIONS_ENABLED=1)
-if(NOT FPGA_BOARD MATCHES ".usm.*" AND NOT DEFINED USM_HOST_ALLOCATIONS_ENABLED)
+if(NOT FPGA_BOARD MATCHES ".usm.*" AND (NOT DEFINED USM_HOST_ALLOCATIONS_ENABLED OR USM_HOST_ALLOCATIONS STREQUAL "0"))
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt:23]: Looking more closely, I'm not sure why you introduced the USM_HOST_ALLOCATIONS variable, as distinct from USM_HOST_ALLOCATIONS_ENABLED?  I recommend removing all references to USM_HOST_ALLOCATIONS (including those below), only using USM_HOST_ALLOCATIONS_ENABLED, and changing this line of code as follows:
```suggestion
if((NOT FPGA_BOARD MATCHES ".usm.*") AND (NOT USM_HOST_ALLOCATIONS_ENABLED))
```

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt [hunk 3]
```diff
@@ -20,16 +20,16 @@ endif()
...
     message(FATAL_ERROR "This tutorial requires a BSP that has USM host allocations enabled.")
 endif()
 
 if (DEFINED USM_HOST_ALLOCATIONS_ENABLED)
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt:27]:
```suggestion
if (USM_HOST_ALLOCATIONS_ENABLED)
```

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt [hunk 4]
```diff
@@ -20,16 +20,16 @@ endif()
 # this tutorial requires USM host allocations. Check the BSP name (which should contain the text 'usm')
 # to ensure the BSP has the required support. Allow the user to define USM_HOST_ALLOCATIONS_ENABLED
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt:21]:
```suggestion
# to ensure the BSP has the required support. Allow the user to define USM_HOST_ALLOCATIONS_ENABLED=1
```

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt [hunk 5]
```diff
@@ -20,16 +20,16 @@ endif()
...
 # to override this check (e.g., cmake .. -DUSM_HOST_ALLOCATIONS_ENABLED=1)
-if(NOT FPGA_BOARD MATCHES ".usm.*" AND NOT DEFINED USM_HOST_ALLOCATIONS_ENABLED)
+if(NOT FPGA_BOARD MATCHES ".usm.*" AND (NOT DEFINED USM_HOST_ALLOCATIONS_ENABLED OR USM_HOST_ALLOCATIONS STREQUAL "0"))
     message(FATAL_ERROR "This tutorial requires a BSP that has USM host allocations enabled.")
 endif()
 
 if (DEFINED USM_HOST_ALLOCATIONS_ENABLED)
   message(STATUS "USM host allocations manually enabled with USM_HOST_ALLOCATIONS_ENABLED=${USM_HOST_ALLOCATIONS_ENABLED}")
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt:28]:
```suggestion
  message(STATUS "USM host allocations manually enabled.")
```

**Review by mdbtucker — COMMENTED** [review]
I believe in CMAKE if you check the value of an undefined variable, it is 0 (false).  This is preferable to checking if a variable is defined, because it prevents the CMAKE warnings where the value of a defined variable is not used.  It also allows you to set something to 0 to indicate you do NOT want to enable that option.  We use this pattern in many other CMAKE files in the samples.

File: DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt (hunk 2 above)
Comment (tyoungsc) [reply DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt]: That was the intention, but I messed it up. I think I have done so now.

**Review by mdbtucker — APPROVED** [review]

**Review by mdbtucker — APPROVED** [review]

**Comment by mdbtucker** [comment]
I want to see the Windows CI run, which is currently blocked by https://jira.devtools.intel.com/projects/REIE/issues/REIE-1347, before merging.

**Pull request merged**